staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -189,6 +194,46 @@ func GetClientIP(req *http.Request) net.IP {
 	return net.ParseIP(req.RemoteAddr)
 }
 
+// Prepares the X-Forwarded-For header for another forwarding hop by appending the previous senders
+// IP address to the X-Forwarded-For chain.
+func PrepareForwardedFor(req *http.Request) {
+	remoteIP := GetRemoteIP(req)
+
+	hdr := req.Header
+	// First check the X-Forwarded-For header for requests via proxy.
+	hdrForwardedFor := hdr.Get("X-Forwarded-For")
+	if hdrForwardedFor != "" {
+		// Append the remote IP if necessary.
```

Comment:
Yeah, for the reason you described below. I couldn't find much on the `X-Real-IP` header, but it looks like most (all?) proxies that set it also set `X-Forwarded-For`